### PR TITLE
Replacing OverflowDB with CPG in-memory Graph

### LINKED ISSUE #218: Remove OverflowDB
We're going to remove OverflowDB as a backing database.

We initially used OverflowDB to persist the CPG in Codyze to storage to minimize memory usage and exhaustion. We still use an early version of OverflowDB and need to update it. The current release is 1.x. This update comes with a reworked API and requires a couple of extensive changes.

We decided to remove OverflowDB because we don't see a big advantage and don't want to invest time in the update.

### PR DESCRIPTION
**Pull request by oxisto**
This PR removes OverflowDB and Gremlin. This is slightly blocked with support for CPG 4.0 (see #184), because otherwise we cannot add new features to the CPG graph, should we need them. That is why this is currently based on a specific CPG commit (using JitPack) that already has the following PRs merged in:
* https://github.com/Fraunhofer-AISEC/cpg/pull/456
* https://github.com/Fraunhofer-AISEC/cpg/pull/458
* https://github.com/Fraunhofer-AISEC/cpg/pull/459
* https://github.com/Fraunhofer-AISEC/cpg/pull/460

This PR deprecates the CLI Jython-based console, because it was unmaintained for a while and the `cpg-console` of the `cpg` project might be a good replacement for it; or rather we could extend the `cpg-console`, but I want to keep this as a separate PR.

Closes #218
Blocked by:
- #184

### CODE REVIEW

**Comment by oxisto** [comment]
The first test in `MarkCppTest` seems to work!

**Comment by oxisto** [comment]
The initial pass is done. @fwendland how should we proceed? Directly remove all the legacy files? I really do not want to keep them too long, they were primarily my reference in building the new evaluator.

File: src/main/java/de/fraunhofer/aisec/analysis/markevaluation/OrderNFAEvaluator.java [hunk 1]
```diff
@@ -494,7 +476,7 @@ private boolean isDisallowedBase(
 		return false;
 	}
 
-	private String getStateSnapshot(Vertex v, HashMap<String, HashSet<Node>> baseToFSMNodes) {
+	private String getStateSnapshot(de.fraunhofer.aisec.cpg.graph.Node v, HashMap<String, HashSet<Node>> baseToFSMNodes) {
 		HashMap<String, HashSet<Node>> simplified = new HashMap<>();
```
Comment (fwendland) [inline src/main/java/de/fraunhofer/aisec/analysis/markevaluation/OrderNFAEvaluator.java:480]: Change `HashMap` to `Map` if possible.

File: src/main/java/de/fraunhofer/aisec/analysis/markevaluation/Evaluator.java [hunk 2]
```diff
@@ -2,53 +2,42 @@
...
+import static de.fraunhofer.aisec.cpg.graph.GraphKt.getGraph;
 
 /**
  * Evaluates all loaded MARK rules against the CPG.
  * <p>
  * returns a number of Findings if any of the rules are violated
  */
 public class Evaluator {
-	private static final Logger log = LoggerFactory.getLogger(Evaluator.class);
+	public static final Logger log = LoggerFactory.getLogger(Evaluator.class);
```
Comment (fwendland) [inline src/main/java/de/fraunhofer/aisec/analysis/markevaluation/Evaluator.java:40]: Why is this one `public`?

**Review by fwendland — APPROVED** [review]
LGTM

**Pull request merged**